mycpp/format_strings.py

```diff
@@ -74,9 +74,11 @@ def Parse(fmt: str) -> List[_Part]:
         if char_code:
             if char_code == '%':
                 part: _Part = LiteralPart('%')
-            else:
+            elif char_code in 'rsdo':
                 part = SubstPart(width, char_code, arg_num)
                 arg_num += 1
+            else:
+                raise AssertionError("mycpp expects a format specifier in [rsdo%%], got '%s' instead" % char_code)
```

Comment:
Let's use RuntimeError here (or some kind of return value)


I try to use AssertionError only for bad code, not bad data

(although mycpp itself might be an exception to that ... it does tend to sloppily use AssertionError instead of actually reporting errors, so I'm glad we are fixing this)